### DISCUSSION TITLE
Generated Java Classes as Serializable

### DISCUSSION THREAD

**Discussion by tarun4279** [Netflix/dgs-framework#430] [General]
We have used Netflix DGS Library & generated certain POJO classes from the graphQL schema

We used those classes in an internal model that needs to be cached and since the DGS  generated classes are no serializable. We see failure in caching for that internal model.

Is it possible to make those generated classes serializable? May be setting certain properties under here

generateJava {
    generateClient = true
    packageName = "com.chegg.${mainPackageName}.generated"
}

Please share if this possible or not? Also, should this be considered as a feature?

**Comment by paulbakker** [comment]
By serializable, do you mean implement the Serializable interface?

Since that’s such an outdated concept in Java, I don’t think it makes a lot
of sense to add it.

**Reply by StringKe** [reply]
Some rpc serialization protocols require the use of `Serializable` Is there any other way to support automatic implementation of Serializable at the moment?

**Comment by paulbakker** [comment]
Serializable is a really outdated concept, I don’t think the generated code
should support it.

**Reply by ivanp** [reply]
It's old, yes. But like [StringKe](https://github.com/StringKe) mentions it is required by many modules, such as Spring Cache default serializer.
